src/gretel_trainer/benchmark/job_spec.py

```diff
@@ -0,0 +1,50 @@
+from dataclasses import dataclass
+from typing import Union
+
+from gretel_trainer.benchmark.core import Dataset
+from gretel_trainer.benchmark.custom.models import CustomModel
+from gretel_trainer.benchmark.gretel.models import GretelModel
+
+
+# TODO(pm): Move this and JobSpec to `core`?
```

Comment:
Seems reasonable to me if you want; I don't have a strong opinion

Reply:
Thanks, I'm going to leave it separate for now then.